spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/procedures/AddFilesProcedure.java

```diff
@@ -265,55 +271,16 @@ public String description() {
     return "AddFiles";
   }
 
-  private void validatePartitionSpec(Table table, Map<String, String> partitionFilter) {
-    List<PartitionField> partitionFields = table.spec().fields();
-    Set<String> partitionNames =
-        table.spec().fields().stream().map(PartitionField::name).collect(Collectors.toSet());
-
-    boolean tablePartitioned = !partitionFields.isEmpty();
-    boolean partitionSpecPassed = !partitionFilter.isEmpty();
-
+  private void nonIdentityPartitionCheck(Table table) {
     // Check for any non-identity partition columns
     List<PartitionField> nonIdentityFields =
-        partitionFields.stream()
+        table.spec().fields().stream()
```

Comment:
Using table.spec() here to determine nonIdentityFields as we want to see if the table has any non-identity partitions until now

Reply:
This would be wrong then, because we are adding files with a different spec id. It's ok to have a current spec with a different partitioning with nonIdentityFields